index.js

```diff
@@ -807,8 +821,23 @@ RingPop.prototype.handleOrProxy =
         if (this.whoami() === dest) {
             return true;
         } else {
-            this.proxyReq(dest, req, res, opts);
+            this.proxyReq(_.extend(opts, {
```

Comment:
would advice against a mutable extension. This might cause suprises for the caller.